File: SuperMovie_server/src/api/Order/GetAllOrder.cs
namespace SuperMovie.Server.Api.Order;

using WebSocketSharp;
using WebSocketSharp.Server;
using Util;

public struct GetAllOrderReq
{
}

public struct OrderRsp
{
    public long OrderId;
    public long OrderUserId;
    public long OrderFilmId;
    public long OrderCinemaId;
    public long OrderScheduleId;
    public string OrderSeat;
    public double OrderPayAmount;
}

public struct GetAllOrderRsp
{
    public List<OrderRsp> Collection;
}

//api : get_all_order
public class GetAllOrder : WebSocketBehavior
{
    protected override void OnMessage(MessageEventArgs e)
    {
        var req = JsonHelper.Parse<GetAllOrderReq>(e.Data);
        var rsp = new GetAllOrderRsp
        {
            Collection = new List<OrderRsp>()
        };
       
        Send(JsonHelper.Stringify(rsp));
    }
}

File: movieui/src/scripts/ws/Order/getAllOrder.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {getAllOrder}
export type {OrderRsp}

const conn = new WebSocket(`${wsUrlRoot}/get_all_order`)

type GetAllOrderReq =
    {}

type OrderRsp =
    {
        OrderId: bigint
        OrderUserId: bigint
        OrderFilmId: bigint
        OrderCinemaId: bigint
        OrderScheduleId: bigint
        OrderSeat: string
        OrderPayAmount: number
    }

type GetAllOrderRsp =
    {
        Collection: OrderRsp[]
    }

async function getAllOrder(req: GetAllOrderReq) {

    const task = recvMsg(conn)
    sendMsg(conn, reqStringify(req)).then()
    const msg = await task

    return <GetAllOrderRsp>rspParse(msg)
}